Sources/Boundaries/Store.swift

```diff
@@ -52,4 +43,85 @@ public final class Store<S, A> {
   public func subscribe<T>(state keyPath: KeyPath<S, T>, _ subscriber: @escaping (T) -> Void) {
     self.subscribe(subscriber <<< view(getting(keyPath)))
   }
+
+  // Runs the effects and dispatches the resulting actions.
+  private func interpret(_ effect: Cmd<E>) {
+    switch effect {
+    case let .execute(e):
+      if let action = self.execute(e) {
+        self.dispatch(action)
+      }
+
+    case .parallel:
+      DispatchQueue.global().async {
+        catOptionals(self.interpretedActions(effect))
+          .forEach { action in
+            DispatchQueue.main.async {
+              self.dispatch(action)
+            }
+        }
+      }
+
+    case let .dispatch(action):
+      self.dispatch(action)
+
+    case let .sequence(effects):
+      effects.forEach(self.interpret)
+    }
+  }
+
+  // Runs the effects and collects all of the resulting actions without dispatching them.
+  private func interpretedActions(_ effect: Cmd<E>) -> [A?] {
+    switch effect {
+    case let .execute(e):
+      return [self.execute(e)]
+
+    case let .parallel(effects):
+      return effects.pmap(self.interpretedActions).flatMap(id)
+
+    case let .dispatch(action):
+      return [action]
+
+    case let .sequence(effects):
+      return effects.flatMap(self.interpretedActions)
+    }
+  }
+}
+
+extension Array {
+  // TODO: move to prelude
+
+  /// Performs a parallel map of a transformation over an array. Splits the array into chunks based on the
+  /// number of computing cores.
+  public func pmap<T>(_ f: @escaping (Iterator.Element) -> T) -> [T] {
+    return self.pmap(queue: .global(), f)
+  }
+
+  public func pmap<T>(queue: DispatchQueue, _ f: @escaping (Iterator.Element) -> T) -> [T] {
```

Comment:
maybe this is overkill lol. maybe it's fine to just `queue.async` for each effect. also hard to imagine someone would emit _that_ many effects...

Reply:
Neat! Probably should benchmark and see if it's necessary. The stack-less IO stuff I was working on seemed to add more overhead than worth, so similar considerations here.

Reply:
yeah gonna revert for this iteration...

Reply:
done, made it simpler by just spawning `queue.async` for each effect